### ISSUE TITLE
Setting property on undefined object valid TypeScript?

### ISSUE THREAD

**Issue by eikooc** [Microsoft/TypeScript#6304]
When writing this in e.g. the TS playground:
```
var Class;
Class.foo = function () {
    return "true";
};
```
There is no error shown. 
Is this intended behavior that the compiler doesn't catch this?

**Comment by kitsonk** [comment]
It is working as designed. The playground runs with `noImplicitAny` set to `false`, which means that the type of `Class` (if you hover over it) is `any` and therefore the second statement is true.  If you did something like:

```typescript
var Class = function () {};
Class.foo = function () {
    return "true";
};
```

You would receive an error, because it will have inferred the type from the assignment.  Implicit `any` is evil, specifically for what you point out there, but obviously would be a huge burden for someone trying to migrate valid JavaScript to TypeScript.

**Comment by DanielRosenwasser** [comment]
As @kitsonk said, you need to either initialize the object so we can infer the type, or add a type annotation.

**Comment by eikooc** [comment]
Thanks for the clarification
